go/mysql/conn.go

```diff
@@ -183,13 +183,38 @@ type Conn struct {
 	// - readEphemeralPacket / recycleReadPacket methods for reads.
 	currentEphemeralPolicy int
 	// TODO (danieltahara): Ultimately get rid of this delineation.
-	currentEphemeralPacket []byte
-	currentEphemeralBuffer *[]byte
+	// currentEphemeralWriteBuffer and currentEphemeralReadBuffer used for tracking
+	// allocated temporary buffers for writes and reads respectively.
+	currentEphemeralWriteBuffer *[]byte
+	currentEphemeralReadBuffer  *[]byte
 }
 
 // bufPool is used to allocate and free buffers in an efficient way.
 var bufPool = sync.Pool{}
 
+// length is always > connBufferSize here, for smaller buffers static buffer field is used
+func getBuf(length int) *[]byte {
+	i := bufPool.Get()
+	if i == nil {
+		buf := make([]byte, length)
+		// shrink it to length
+		buf = buf[:length]
+		return &buf
+	}
+	// We got an array from the pool, see if it's
+	// big enough.
+	buf := i.(*[]byte)
+	if cap(*buf) >= length {
+		// big enough, shrink to length and use it.
+		*buf = (*buf)[:length]
+		return buf
+	}
```

Comment:
I think we should `Put` the buffer back at this point. This way, someone else may benefit from this already allocated slice.